pkg/querier/querier.go

```diff
@@ -535,6 +535,12 @@ func (q *Querier) validateQueryRequest(ctx context.Context, req *logproto.QueryR
 			"max streams matchers per query exceeded, matchers-count > limit (%d > %d)", len(matchers), maxStreamMatchersPerQuery)
 	}
 
+	maxEntriesLimit := q.limits.MaxEntriesLimitPerQuery(userID)
```

Comment:
Same as for the frontend we want to apply limits only to log queries. Not sure what's the best way for that here. Grafana does send the limit even for metrics .... and we should not apply limit. WDYT ?